src/rezplugins/shell/csh.py

```diff
@@ -157,8 +159,8 @@ def _saferefenv(self, key):
         self._addline("if (!($?%s)) setenv %s" % (key, key))
 
     def setenv(self, key, value):
-        value = self.escape_string(value, is_path=self._is_pathed_key(key))
-        self._addline('setenv %s %s' % (key, value))
+        new_value = self.escape_string(value, is_path=self._is_pathed_key(key))
+        self._addline('setenv %s %s' % (key, new_value))
```

Comment:
Should we also normalize the source command? (I can't comment on non modified code...)